### DISCUSSION TITLE
Grain boundary fundamental zone plot in MTeX

### DISCUSSION THREAD

**Discussion by SagarChandra** [mtex-toolbox/mtex#672] [Ask Anything] [answered]
Hi all,

I would like to congratulate the developers of MTeX for such a nice and useful code!

I have a general question; I am using MTeX 5.5.2. Is it possible to construct grain boundary fundamental zones in MTeX. For example, for FCC crystals, the fundamental zone of a sigma3 CSL boundary would be represented by a steriographic triangle having {111}, {110} and {112} as its three vertices.  

What I want to do is to plot the variations of GB properties, like GB energy, for certain sigma boundaries. I believe that the best way to show the variation in GB properties with change in GB plane normal (for the same CSL boundary) is to show color contours of that property in the corresponding CSL fundamental zone plot.  Is it possible in MTeX.

Best Regards,
Sagar

**Comment by ralfHielscher** [comment]
Hi Sagar,

I'm a bit unsure what you are talking about. There exist 
 * fundamental zones as part of the misorientation space describing orientation relationships at grain boundaries. These fundamental zones are three dimensional subspaces of the space of all misorientations see https://mtex-toolbox.github.io/OrientationFundamentalRegion.html
 * fundamental sectors are subsets of a sphere and are used to describe e.g. inverse pole figures or rotational axes.

So maybe you want the describe boundary misorientations by the misorientation axis and plot it into a spherical triangle?

Ralf.

**Comment by SagarChandra** [comment]
Hi Ralf,

I am talking about the plot which is a stereographic projection of a portion of a sphere, where Σ3 GBs, for example, with boundary plane orientations inside this region have unique character. Please see the following link which introduces such GB fundamental zones. 

https://www.nature.com/articles/srep15476

Is there a way to plot these in MTeX?


Best,
Sagar

**Comment by ralfHielscher** [comment, marked as answer]
Hi Sagar,

the following lines produces such an triangle.
```matlab
cs = crystalSymmetry('432'); % define crystal symmetry
v = Miller({1,1,1},{2,-1,-1},{1,0,-1},cs) % the vertices
sR = sphericalRegion.triangle(v(1),v(2),v(3)) % define the triangle
% rotate the triangle such that (111) is becomes the center
rot = rotation.map(v(1),zvector)
% plot the rotated triangle
plot(rotate(sR,rot))
% annotate the vertices
annotate(rot*v,'label',char(v,'cell','latex'),'backgroundColor','white')
```
<img width="369" alt="tmp" src="https://user-images.githubusercontent.com/8619241/103529603-3d8a7380-4e86-11eb-8397-4b5c628d7a08.png">
Now you can plot arbitrary directions `d` into the triangle by

```matlab
plot(rot * d)
```

Ralf.

**Comment by SagarChandra** [comment]
Hi Ralf,

Thank you very much for your reply, I will definitely give it a try.
Now, if I want to plot the specific orientations in the triangle based on
coloring (contours or discrete points) as per the GB energies (already
available for each orientation vector), how to go about it using MTeX?

I thank you once again for your time and guidance.

Regards.

**Reply by ralfHielscher** [reply]
From the paper you attached I understand that they plotted the boundary plane normals into the triangle - not orientations.
Hence you need to figure out how to compute these boundary plane normals.

Ralf.

**Comment by SagarChandra** [comment]
Hi Ralf,

Yes, sorry if the word orientations was misleading. I am having these
boundary plane normals for each boundary along with the corresponding GB
energy of each boundary. Now, I want to show the variations of these GB
energies with boundary plane normal in the triangle. If required, I can
also share a sample text file here containing this information.

Thank you.

**Comment by SagarChandra** [comment]
Hi Ralf,

Thank you very much for the script lines above; the zone plots are coming as desired.

Best regards,
sagar
